### PR TITLE
Logstash: Multitenant

### PR DESCRIPTION
**Pull request by RosesAreR3d**

### CODE REVIEW

File: clients/cmd/logstash/Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,165 @@
+PATH
+  remote: logstash/logstash-core-plugin-api
```
Comment (cyriltovena) [inline clients/cmd/logstash/Gemfile.lock:2]: Should the lock file be committed ? I'm not a ruby expert.

File: clients/cmd/logstash/README.md [hunk 2]
```diff
@@ -1,4 +1,21 @@
-# Contributing to Loki Logstash Output Plugin
+# Fork of Loki Logstash Output Plugin
```
Comment (cyriltovena) [inline clients/cmd/logstash/README.md:1]: I think the doc was meant for a fork so you should adjust this please.

File: loki-build-image/.vs/VSWorkspaceState.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (cyriltovena) [inline loki-build-image/.vs/VSWorkspaceState.json:1]: Let's remove this folder.

File: clients/cmd/logstash/Dockerfile [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
-
-USER logstash
-ENV PATH /usr/share/logstash/vendor/jruby/bin:/usr/share/logstash/vendor/bundle/jruby/2.5.0/bin:$PATH
-ENV LOGSTASH_PATH="/usr/share/logstash"
-ENV GEM_PATH /usr/share/logstash/vendor/bundle/jruby/2.5.0
-ENV GEM_HOME /usr/share/logstash/vendor/bundle/jruby/2.5.0
-
-RUN gem install bundler:2.1.4
-
-COPY --chown=logstash:logstash ./clients/cmd/logstash/ /home/logstash/
```
Comment (cyriltovena) [inline clients/cmd/logstash/Dockerfile:11]: is it fine to remove this ? Do you think it's not used ?

**Review by cyriltovena — COMMENTED** [review]
It's not really clear  but I think you can get the tenant from a global config or from labels in the message in which case you're not removing it from the labels.

I think it's worth mentioning in the docs.

File: clients/cmd/logstash/Dockerfile (hunk 4 above)
Comment (marekasf) [reply clients/cmd/logstash/Dockerfile:11]: please do not remove it.

File: clients/cmd/logstash/Dockerfile [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
+RUN bundle install --path=/usr/share/logstash/vendor/bundle && \
+    bundle exec rake vendor && \
+    bundle exec rspec
+
+RUN cat logstash-output-loki.gemspec | grep s.version | awk '{print $3}' |  cut -d "'" -f 2 > VERSION
+
+RUN gem build logstash-output-loki.gemspec && \
+    PLUGIN_VERSION=$(cat VERSION); /usr/share/logstash/bin/logstash-plugin install logstash-output-loki-${PLUGIN_VERSION}.gem
+
+EXPOSE 5044
```
Comment (marekasf) [inline clients/cmd/logstash/Dockerfile:24]: The file should remain oruginal

File: clients/cmd/logstash/Gemfile.lock (hunk 1 above)
Comment (marekasf) [reply clients/cmd/logstash/Gemfile.lock:2]: it should not be commited

File: clients/cmd/logstash/README.md (hunk 2 above)
Comment (marekasf) [reply clients/cmd/logstash/README.md:1]: Please add a separate section with a description of the new plugin features

File: clients/cmd/logstash/Rakefile [hunk 6]
```diff
@@ -1 +1,2 @@
 require "logstash/devutils/rake"
+require 'bundler/gem_tasks'
```
Comment (marekasf) [inline clients/cmd/logstash/Rakefile:2]: it should not be added

File: loki-build-image/.vs/VSWorkspaceState.json (hunk 3 above)
Comment (marekasf) [reply loki-build-image/.vs/VSWorkspaceState.json:1]: This file should not be added

**Review by marekasf — CHANGES_REQUESTED** [review]

**Pull request closed without merging**